aquamarine/src/particle_executor.rs

```diff
@@ -72,109 +71,127 @@ impl<RT: AquaRuntime> ParticleExecutor for RT {
+            }
+            let stats = InterpretationStats::failed();
+            let effects = ParticleEffects::empty();
+            FutResult {
+                // We loose an AVM instance here
+                // But it will be recreated via VmPool
+                runtime: None,
+                effects,
+                stats,
+            }
+        }
+    }
+}
 
-                    FutResult {
-                        runtime: Some(self),
-                        effects,
-                        stats,
-                    }
-                })
-            })
-            .expect("Could not spawn 'Particle' task");
+#[instrument(level = tracing::Level::INFO, skip_all)]
+async fn process_avm_result<RT>(
+    data_store: Arc<ParticleDataStore>,
+    current_peer_id: PeerId,
+    prev_data_len: usize,
+    avm_result: AVMCallResult<'_, RT>,
+) -> AVMRes<RT>
+where
+    RT: AquaRuntime,
+{
+    let particle_id = avm_result.particle.id;
+    let stats = avm_result.stats;
+    match &avm_result.avm_outcome {
+        Ok(outcome) => {
+            let len = outcome.data.len();
+            tracing::trace!(
+                target: "execution", particle_id = particle_id,
+                "Particle interpreted in {} [{} bytes => {} bytes]",
+                humantime::format_duration(stats.interpretation_time), prev_data_len, len
+            );
 
-        async move {
-            let result = task.await;
-            match result {
-                Ok(res) => res,
-                Err(err) => {
-                    if err.is_cancelled() {
-                        tracing::warn!(particle_id, "Particle task was cancelled");
-                    } else {
-                        tracing::error!(particle_id, "Particle task panic");
-                    }
-                    let stats = InterpretationStats {
-                        interpretation_time: Duration::ZERO,
-                        call_time: Duration::ZERO,
-                        new_data_len: None,
-                        success: false,
-                    };
-                    let effects = ParticleEffects::empty();
-                    FutResult {
-                        // We loose an AVM instance here
-                        // But it will be recreated via VmPool
-                        runtime: None,
-                        effects,
-                        stats,
-                    }
+            if data_store.detect_anomaly(stats.interpretation_time, stats.memory_delta, outcome) {
+                let anomaly_result = data_store
+                    .save_anomaly_data(
+                        avm_result.particle.script.as_str(),
+                        &avm_result.particle.data,
+                        &avm_result.call_results,
+                        &avm_result.particle_params,
+                        outcome,
+                        stats.interpretation_time,
+                        stats.memory_delta,
+                    )
```

Comment:
```suggestion
                    .save_anomaly_data(
                        &avm_result
                        outcome,
                        stats
                    )
```

would be so much shorter

Reply:
well make it one, why not?